### ISSUE TITLE
Fonts Library: Fix variants count color contrast and l10n best practices

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58110]
### Description

The font variants count in the side panel has a very low color contrast ratio of 2.68:1

![Screenshot 2024-01-23 at 10 46 31](https://github.com/WordPress/gutenberg/assets/1682452/107a15ba-67f0-4b87-a7fb-7fb670bf0032)

Also, concatenation with variables and translatable strings is not OK and doesn't meet the WordPress l10n best practices, even ore in this case where `_n()` is in use.

https://github.com/WordPress/gutenberg/blob/c6ac9c010551aed63a5e31f12b0796b08e941747/packages/edit-site/src/components/global-styles/font-family-item.js#L35-L38

Wirth also mentioning that, as far as I can tell, the gray `#9e9e9e` is not part of the established color palette. Such usage of unique, ad-hoc, colors shouldn't be allowed and, in any case, a sufficient color contrast ratio is a basic accessibility requirement. Also, inline styles should be avoided.

Introduced in https://github.com/WordPress/gutenberg/pull/53884

### Step-by-step reproduction instructions

- Go to Site editor > Styles > Typography
- Observe the text that indicates the font variants count has a very low color contrast ratio.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
